clearnode/rpc_router_public.go

```diff
@@ -297,3 +314,48 @@ func (r *RPCRouter) HandleGetLedgerEntries(c *RPCContext) {
 	c.Succeed(req.Method, resp)
 	logger.Info("ledger entries retrieved", "accountID", accountID, "asset", asset, "wallet", walletAddress)
 }
+
+func (r *RPCRouter) HandleGetTransactions(c *RPCContext) {
+	ctx := c.Context
+	logger := LoggerFromContext(ctx)
+	req := c.Message.Req
+
+	var params GetTransactionsParams
+	if err := parseParams(req.Params, &params); err != nil {
+		c.Fail(err.Error())
+		return
+	}
+
+	var txType *TransactionType
+	if params.TxType != "" {
+		parsedType, ok := ParseTransactionType(params.TxType)
+		if !ok {
+			c.Fail("invalid transaction type")
+			return
+		}
+		txType = &parsedType
+	}
+
+	transactions, err := GetTransactions(r.DB, params.AccountID, params.Asset, txType)
+	if err != nil {
+		logger.Error("failed to get transactions", "error", err)
+		c.Fail("failed to get transactions")
+		return
+	}
+
+	resp := make([]TransactionResponse, len(transactions))
+	for i, tx := range transactions {
+		resp[i] = TransactionResponse{
+			TxHash:      tx.Hash,
+			TxType:      tx.Type.String(),
+			FromAccount: tx.FromAccount,
+			ToAccount:   tx.ToAccount,
+			Asset:       tx.AssetSymbol,
+			Amount:      tx.Amount,
+			CreatedAt:   tx.CreatedAt,
```

Comment:
Maybe it is a good idea to create a `TransactionResponse::FromTransaction(...)` method to create transaction response from transaction struct?